From 7a6f48c2a96acca0a2426cabebfc50302478697e Mon Sep 17 00:00:00 2001
From: Peter Hutterer <peter.hutterer@who-t.net>
Date: Thu, 28 Oct 2010 14:09:40 +1000
Subject: [PATCH 07/10] Hack in ignorebutton handling.

Since it got switched to button 0, this caused an error (wacomcpl treats the
ignorebutton as 34). So for buttons 0, use $ignoreButton instead.

Signed-off-by: Peter Hutterer <peter.hutterer@who-t.net>
---
 wacomcpl-exec |    6 +++++-
 1 files changed, 5 insertions(+), 1 deletions(-)

diff --git a/wacomcpl-exec b/wacomcpl-exec
index a44b66a..35ee7b0 100755
--- a/wacomcpl-exec
+++ b/wacomcpl-exec
@@ -1080,8 +1080,12 @@ proc defaultButton {} {
     global db11 db12 db13 db14 isLCD
     global dm sm dmv smv numButton startS numStrips spName
     global device getOptionDefault getDeviceModel
+    global ignoreButton
+
     for { set i 1 } { $i <= $numButton } { incr i 1 } {
-	set db$i $db($getOptionDefault($device,$spName($i)))
+	set btn $getOptionDefault($device,$spName($i))
+	if { $btn == 0 } { set btn $ignoreButton }
+	set db$i $db($btn)
     }
 
     if { $startS } {
-- 
1.7.1

